### PR TITLE
build.d: Change  working directory to the project root

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
.. instead of `src`. This ensures that error messages, coverage data, ... will be reported relative to the root directory as expected by many IDE's.

### CODE REVIEW

**Comment by MoonlightSentinel** [comment]
This breaks CodeCov ...

File: .circleci/run.sh [hunk 1]
```diff
@@ -193,9 +193,9 @@ codecov()
 {
     # CodeCov gets confused by lst files which it can't matched
     rm -rf test/runnable/extra-files
+    find . -name '*.lst' -exec echo {} \; -exec  head {} \;
     download "https://codecov.io/bash" "https://raw.githubusercontent.com/codecov/codecov-bash/master/codecov" "codecov.sh"
-    cd src # need to run from compilation folder for gcov to find sources
-    bash ../codecov.sh -p .. -x gcov-4.9 # must match g++ version (on CircleCI `g++` is 4.9 and `gcov` 4.6 :/)
+    bash codecov.sh -p . -s src -Z -x gcov-4.9 # must match g++ version (on CircleCI `g++` is 4.9 and `gcov` 4.6 :/)
```
Comment (wilzbach) [inline .circleci/run.sh:198]: We don't have C or C++ files anymore, so you could drop the gcov flag here.

**Review by wilzbach — COMMENTED** [review]
In mars.d the coverage base directory is set. I would guess that this needs adjustment too.

**Comment by MoonlightSentinel** [comment]
Thanks, CodeCov works again

**Comment by Geod24** [comment]
Wild guess: the shenanigans in `mars.d` predate https://github.com/dlang/druntime/pull/1916 and could be replaced now.

File: .circleci/run.sh [hunk 2]
```diff
@@ -194,8 +194,7 @@ codecov()
     # CodeCov gets confused by lst files which it can't matched
     rm -rf test/runnable/extra-files
     download "https://codecov.io/bash" "https://raw.githubusercontent.com/codecov/codecov-bash/master/codecov" "codecov.sh"
-    cd src # need to run from compilation folder for gcov to find sources
-    bash ../codecov.sh -p .. -x gcov-4.9 # must match g++ version (on CircleCI `g++` is 4.9 and `gcov` 4.6 :/)
```
Comment (Geod24) [inline .circleci/run.sh:198]: What's the story here ?

Comment (MoonlightSentinel) [reply .circleci/run.sh:198]: Changing the working directory implies different paths for coverage files.
`dmd_expressionsem.lst` located in `src` becomes `src_dmd_expressionsem.d` in the root directory.

See https://github.com/dlang/dmd/pull/11534#discussion_r467500690 regarding `gcov`

**Comment by MoonlightSentinel** [comment]
> Wild guess: the shenanigans in `mars.d` predate [dlang/druntime#1916](https://github.com/dlang/druntime/pull/1916) and could be replaced now.

Probably, but embedding `rt_options` might be better suited to keep the current behaviour?

**Review by Geod24 — APPROVED** [review]

**Pull request merged**